tests/filecheck/transforms/individual_rewrite/add-same.mlir

```diff
@@ -0,0 +1,18 @@
+// RUN:xdsl-opt %s --split-input-file -p 'apply-individual-rewrite{matched_operation_index=2 operation_name="arith.addi" pattern_name="AdditionOfSameVariablesToMultiplyByTwo"}'| filecheck %s
+
+
+// CHECK:      %[[#var:]] = "test.op"() : () -> i32
+// CHECK-NEXT: %[[#two:]] = arith.constant 2 : i32
+// CHECK-NEXT: %{{.*}} = arith.muli %[[#var]], %[[#two]] : i32
+
+%0 = "test.op"() : () -> (i32)
+%1 = arith.addi %0, %0 : i32
+
+// -----
+
+// CHECK:      %[[#var:]] = "test.op"() : () -> i1
+// CHECK-NEXT: %[[#zero:]] = arith.constant false
+// CHECK-NEXT: %{{.*}} = arith.muli %[[#var]], %[[#zero]] : i1
+
+%0 = "test.op"() : () -> (i1)
+%1 = arith.addi %0, %0 : i1
```

Comment:
nit: are the regexes here really more readable than giving names to ssa values and checking for those directly?

Reply:
I've removed 1, but can't/haven't named the constants

Reply:
If you name the inputs, the constants they are replaced with inherit their hints, but this is more of a nit at this point

Reply:
I don't want the test to be checking that the constant has the same name as the original operation, and so would prefer to use the regex